build.sbt

```diff
@@ -30,42 +34,37 @@ lazy val migrator = (project in file("migrator")).settings(
     "-Xmx2048M",
     "-XX:MaxPermSize=2048M",
     "-XX:+CMSClassUnloadingEnabled"),
-  scalacOptions ++= Seq("-deprecation", "-unchecked", "-Ypartial-unification"),
   Test / parallelExecution := false,
   fork                     := true,
   scalafmtOnCompile        := true,
   libraryDependencies ++= Seq(
     "org.apache.spark" %% "spark-streaming"      % sparkVersion % "provided",
     "org.apache.spark" %% "spark-sql"            % sparkVersion % "provided",
-    "com.amazonaws"    % "aws-java-sdk-dynamodb" % awsSdkVersion,
-    "com.amazonaws"    % "aws-java-sdk-s3"       % awsSdkVersion,
-    "com.amazonaws"    % "aws-java-sdk-sts"      % awsSdkVersion,
-    ("com.amazonaws" % "dynamodb-streams-kinesis-adapter" % dynamodbStreamsKinesisAdapterVersion)
-      .excludeAll(InclExclRule("com.fasterxml.jackson.core")),
-    "com.amazon.emr" % "emr-dynamodb-hadoop" % "4.16.0",
-    "io.circe"       %% "circe-generic"      % "0.11.1",
-    "io.circe"       %% "circe-parser"       % "0.11.1",
-    "io.circe"       %% "circe-yaml"         % "0.10.1",
+    ("org.apache.hadoop" % "hadoop-aws"           % hadoopVersion) // Note: this package still depends on the AWS SDK v1
+      // Exclude the AWS bundle because it creates many conflicts when generating the assembly
+      .excludeAll(
+        InclExclRule("com.amazonaws", "aws-java-sdk-bundle"),
+      ),
+    "software.amazon.awssdk"    % "s3-transfer-manager" % awsSdkVersion,
+    "software.amazon.awssdk"    % "dynamodb" % awsSdkVersion,
+    "software.amazon.awssdk"    % "s3"       % awsSdkVersion,
+    "software.amazon.awssdk"    % "sts"      % awsSdkVersion,
+    "com.datastax.spark" %% "spark-cassandra-connector" % "3.1.0-49-g38c045a4",
```

Comment:
this needs to be dynamic based on submodule build, is that possible? maybe in spark connector we'd need to switch to different publish group?

Reply:
> this needs to be dynamic based on submodule build, is that possible?

It might work if we define the `spark-cassandra-connector` as a source dependency, but source dependencies are not very well supported by sbt, so I don’t think we can really do better than manually updating the dependency version.